src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -573,6 +573,12 @@ private IMethodSymbol Instantiate(IMethodSymbol method, IList<ITypeSymbol> invoc
                 }
 
                 var typeArguments = method.ConstructedFrom.TypeParameters.Select(tp => bestMap.GetValueOrDefault(tp) ?? tp).ToArray();
+
+                if (!ReferenceEquals(method, method.ConstructedFrom))
+                {
+                    return method.ConstructedFrom.Construct(typeArguments);
```

Comment:
Just do this regardless?  If they're ReferenceEquals, it will be equivalent, right?